fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ExpressionAnalyzer.java

```diff
@@ -407,8 +407,14 @@ public Expression visitUnboundFunction(UnboundFunction unboundFunction, Expressi
         Pair<? extends Expression, ? extends BoundFunction> buildResult = builder.build(functionName, arguments);
         buildResult.second.checkOrderExprIsValid();
         Optional<SqlCacheContext> sqlCacheContext = Optional.empty();
+        StatementContext statementContext = context.cascadesContext.getStatementContext();
+
+        // If it is prepared statement in EXECUTE phase, record nonDeterministic info
+        if (ConnectContext.get() != null && ConnectContext.get().getCommand() == MysqlCommand.COM_STMT_EXECUTE
+                && !buildResult.second.isDeterministic()) {
+            statementContext.setHasNondeterministicInPreparedStatement(true);
+        }
```

Comment:
set this flag for all statement or mv this flag out of statement conctext

Reply:
done